catalog/lib/theme/app_dimensions.dart

```diff
@@ -26,4 +28,26 @@ class AppDimens {
         spacing20: 20,
         spacing24: 24,
       );
+
+  @override
+  AppDimension copyWith({MaterialColor? primary}) =>
+      AppDimension.getDefaultDimensions();
+
+  @override
+  AppDimension lerp(AppDimension? other, double t) {
+    if (other is! AppDimension) {
+      return this;
+    }
+    return AppDimension(
+      spacing2: _lerpInt(spacing2, other.spacing2, t).toInt(),
+      spacing4: _lerpInt(spacing4, other.spacing4, t).toInt(),
+      spacing8: _lerpInt(spacing8, other.spacing8, t).toInt(),
+      spacing12: _lerpInt(spacing12, other.spacing12, t).toInt(),
+      spacing16: _lerpInt(spacing16, other.spacing16, t).toInt(),
+      spacing20: _lerpInt(spacing20, other.spacing20, t).toInt(),
+      spacing24: _lerpInt(spacing24, other.spacing24, t).toInt(),
+    );
+  }
+
+  double _lerpInt(int a, int b, double t) => a + (b - a) * t;
```

Comment:
What do you think about changing the names so it's more readable, like a = oldValue, b = newValue and t = factor